tests/test_streaming_scenario.cpp

```diff
@@ -91,7 +91,7 @@ void insert_next_batch(diskann::Index<T, TagT, LabelT> &index, size_t start, siz
         std::cout << std::endl << "Inserting from " << start << " to " << end << std::endl;
 
         size_t num_failed = 0;
-#pragma omp parallel for num_threads(insert_threads) schedule(dynamic) reduction(+ : num_failed)
+#pragma omp parallel for num_threads((int32_t)insert_threads) schedule(dynamic) reduction(+ : num_failed)
```

Comment:
why (signed) int32_t for num_threads?

Reply:
GCC doesn't complain about it, but MSVC gives the following warning (which is treated as an error):

> ...\microsoft\DiskANN\tests\test_streaming_scenario.cpp(94): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data

I can't find the relevant OpenMP documentation that defines the expected integer type for the `num_threads` clause. These two sources seem reliable, but they don't go beyond describing the expected type as "num" or "integer":
* https://learn.microsoft.com/en-us/cpp/parallel/openmp/reference/openmp-clauses?view=msvc-170#num-threads
* https://www.openmp.org/spec-html/5.0/openmpse14.html#x54-800002.6

To appease the MSVC, I'm inclined to cast it to the signed int as warned. We should never be in a situation where we are declaring billions of threads, after all. And we can leave the input data type as size_t since that is more explicit as to the expected behavior.